test/fw/ptl/utils/pbs_snaputils.py

```diff
@@ -0,0 +1,1777 @@
+            self.anon_obj.anonymize_file(file_path, inplace=True)
+
+    def __copy_dir_with_core(self, host, src_path, dest_path, core_dir,
+                             except_list=[], only_core=False):
+        """
+        Copy over a directory recursively which might have core files
+        When a core file is found, capture the stack trace from it
+
+        :param host: name of the host where the source is located
+        :type str
+        :param src_path: path of the source directory
+        :type str
+        :param dest_path: path of the destination directory
+        :type str
+        :param core_dir: path to the directory to store core files' trace
+        :type str
+        :param except_list: list  of files/directories (basenames) to exclude
+        :type list
+        :param only_core: Copy over only core files?
+        :type bool
+        """
+        dir_list = None
+        try:
+            dir_list = self.du.listdir(host, src_path, fullpath=False,
+                                       sudo=self.sudo)
+        except Exception:
+            self.logger.info("Can't find/access " + src_path + " on host " +
+                             host)
+            return
+
+        if dir_list is None:
+            return
+
+        # Go over the list and copy over everything
+        # If we find a core file, we'll store backtrace from it inside
+        # core_file_bt
+        if self.du.is_localhost(host) is False:
+            prefix = host + ":"
+        else:
+            prefix = ""
+
+        for item in dir_list:
+            if item in except_list:
+                continue
+
+            item_src_path = os.path.join(src_path, item)
+            if only_core is False:
+                item_dest_path = os.path.join(dest_path, item)
+            else:
+                item_dest_path = core_dir
+
+            if self.du.isdir(host, item_src_path, sudo=self.sudo):
+                # Make sure that the directory exists in the snapshot
+                if self.du.isdir(item_dest_path) is False:
+                    # Create the directory
+                    os.makedirs(item_dest_path, 0755)
+                # Recursive call to copy contents of the directory
+                self.__copy_dir_with_core(host, item_src_path, item_dest_path,
+                                          core_dir, except_list, only_core)
+            else:
+                # Copy the file over
+                item_src_path += prefix
+                self.du.run_copy(src=item_src_path, dest=item_dest_path,
+                                 recursive=False, mode=0755,
+                                 level=logging.DEBUG, sudo=self.sudo)
+
+                # Check if this is a core file
+                # If it is then this method will capture its stack trace
+                is_core = self.__evaluate_core_file(item_dest_path, core_dir)
+                if is_core is True:
+                    continue
+
+                if only_core is True:
```

Comment:
No actually, it is needed. I added some comments to explain it now. If 'only_core' is True and the file that was copied over is not a core file, then we need to delete it and NOT add it to archive. So, even though add_to_archive will delete it, it will do so after having already added the file to the tarball, and we don't want that if that 'if' condition is True